src/protocol/mod.rs

```diff
@@ -149,6 +149,12 @@ enum WriteState {
     WriteData(VecWriter),
 }
 
+enum NextWriteState {
+    Same,
+    None,
```

Comment:
`Done` doesn't make sense. That's the reason the next state is "nothing", not the state itself.

`Nothing` is goofy but I guess it's fine...